### PR TITLE
Initial python linter

### PR DESCRIPTION
**Pull request by nateinaction**
Hey team,

This PR adds
- a python linter called [Black](https://pypi.org/project/black/)
- README update with instructions on how to install black locally
- a Makefile with initial automatic formatter tooling `make fmt`
- our first CI pipeline ensuring that future pull requests pass some base tests

This looks like a big PR but all python modifications were performed by Black and were not made by a human.

In the future I'd like to implement more advanced static analysis that will help us detect and prevent bugs as well as automated testing frameworks for both the python and cpp codebases.

When this PR is merged, I'd like to restrict the `main` branch from being merged to directly. All changes will require PRs and all PRs will need to pass testing.

I also suggest that we consider blocking pull requests on reviews. Pull requests can serve as documentation for why a change was made and reviews can ensure that at least one other person knows what the change does.

### CODE REVIEW

**Review by Mikefly123 — APPROVED** [review]
LGTM! 

Ran an initial test of everything on my machine with a satellite attached and all of the code appears to run without issues.

File: README.md [hunk 1]
```diff
@@ -47,3 +47,9 @@ This software contains all of the libraries required to operate the sensors, pys
 - **pysquared_rfm9x.py** This is a library that implements all the radio hardware. This code is a modified version of the pycubed_rfm9x which is a modified version of the adafruit_rfm9x file.
 ## tests
 This software is used for performing tests on the satellite
+
+## Linting setup
+
+1. Create your venv `python3 -m venv venv`
+2. Activate your venv `source ./venv/bin/activate`
+3. Install required packages `pip install -r requirements.txt`
```
Comment (Mikefly123) [inline README.md:55]: Recommending adding a line here: 
```markdown
4. Run Black using `black {source_file_or_directory}`
```

**Comment by Mikefly123** [comment]
Your recommendations on restricting merging to main on reviewed Pull Requests sounds good to me as well! 

I have also been meaning to setup a real versioning system for our software as well. Probably out of the scope of this PR but we can look at that in the next few days!

File: README.md (hunk 1 above)
Comment (nateinaction) [reply README.md:55]:
```suggestion
3. Install required packages `pip install -r requirements.txt`
4. Run the automatic formatter with `make fmt`
```

**Comment by nateinaction** [comment]
@Mikefly123 Thanks for being open to these changes! I've implemented the branch rules to restrict changes to pull requests and ensure that the lint check is passing. I'll hold off on requiring peer reviews until we have a chance to talk about it in the next software meeting.

**Pull request merged**